legion/engine/physics/systems/physicssystem.cpp

```diff
@@ -21,10 +21,22 @@ namespace legion::physics
 
     void PhysicsSystem::setup()
     {
-        createProcess<&PhysicsSystem::fixedUpdate>("Physics", m_timeStep);
-
         m_broadPhase = std::make_unique<BroadphaseUniformGridNoCaching>(math::vec3(3, 3, 3));
+    }
+
+    void PhysicsSystem::update(legion::time::span deltaTime)
+    {
+        m_accumulator += deltaTime;
 
+        int currentFrameFixedUpdateCount = 0;
+
+        while (m_accumulator > m_timeStep && currentFrameFixedUpdateCount < m_maxTimeStepPerFrame)
```

Comment:
if `currentFrameFixedUpdateCount` (which is a little overly descriptive) is mainly used to compare against `m_maxTimeStepPerFrame` (maybe m_maxIntervals?) they should be of the same type. at the very least of the same signedness.